tripl/tripl.py

```diff
@@ -583,27 +588,29 @@ def _entity_lookup(self, lookup):
             # we look up the index, and see if the vals pointed to intersect with the lookup val for each key in
             # the pattern
             lookup_vals = val if isinstance(val, (list, set)) else [val]
-            return set(eid for eid, vals in self._aev_index.get([attr]).keys.items() if vals.intersection(lookup_vals))
+            return set(eid for eid, vals in self._aev_index.get([attr]).keys.items()
+                       if vals.intersection(lookup_vals))
 
     def match(self, pattern):
-        xf_subpattern = lambda v: (self.match(v) if isinstance(v, dict) else v)
+        def xf_subpattern(v):
+            return self.match(v) if isinstance(v, dict) else v
+
         pattern = {k: xf_subpattern(v) for k, v in pattern.items()}
-        return reduce(set.intersection, map(self._entity_lookup, pattern.items()))
+        return functools.reduce(set.intersection, map(self._entity_lookup, pattern.items()))
```

Comment:
PyCharm notices a type mismatch here (it expected a function `(T, T) -> T` but got something that goes from `T` to a set of `T`), but I was unable to figure out whether this is a false positive or if it is a bug, how it should actually be.

Reply:
The warning can be silenced by replacing the line with this:
```python
        def intersection(a, b):
            return a.intersection(b)

        return functools.reduce(intersection, map(self._entity_lookup, pattern.items()))
```

Could it be that `set.intersection` cannot / should not be used as a static method in Python 3 anymore or is this a false alarm?

Reply:
I think it's probably a false alarm (PyCharm outsmarting itself). IMHO, it's one of the nicer features of pythons that methods are just functions, and that you can do this sort of functional thing when it suits.